lib/libxdp/xsk.c

```diff
@@ -306,24 +308,43 @@ static int xsk_create_umem_rings(struct xsk_umem *umem, int fd,
 	return err;
 }
 
-int xsk_umem__create_with_fd(struct xsk_umem **umem_ptr, int fd,
-			     void *umem_area, __u64 size,
-			     struct xsk_ring_prod *fill,
-			     struct xsk_ring_cons *comp,
-			     const struct xsk_umem_config *usr_config)
-{
-	struct xdp_umem_reg mr;
+struct xsk_umem *xsk_umem__create_opts(struct xsk_umem_opts *opts) {
+	int err, fd;
+	void *umem_area;
+	__u64 size;
+	struct xsk_ring_prod *fill;
+	struct xsk_ring_cons *comp;
+	const struct xsk_umem_config *usr_config;
 	struct xsk_umem *umem;
-	int err;
+	struct xdp_umem_reg mr;
+	struct xdp_umem_reg_v1 mr_v1;
```

Comment:
These could just be in the blocks of the respective if statement branches below, then they can also be initialised by a `= {}` assignment